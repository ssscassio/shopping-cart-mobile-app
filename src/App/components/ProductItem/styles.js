import colors from '../../config/colors';

const shadowStyle = {
  shadowOpacity: 0.35,
  shadowOffset: {
    width: 0,
    height: 2,
  },
  shadowColor: '#000',
  shadowRadius: 3,
  elevation: 5,
};

export default {
  container: {
    alignItems: 'center',
    marginVertical: 10,
    marginHorizontal: 20,
    backgroundColor: colors.backgroundLight,
    flex: 1,
    padding: 10,
    flexDirection: 'row',
    justifyContent: 'space-between',
    ...shadowStyle,
  },
  picture: {
    height: 100,
    width: 100,
    backgroundColor: colors.background,
    borderRadius: 5,
    borderWidth: 2,
    borderColor: colors.secondary,
  },
  rightContainer: {
    flexDirection: 'column',
  },
  productTitle: {
    color: '#333',
    fontWeight: 'bold',
    fontSize: 15,
  },
  productPrice: {
    color: colors.primary,
  },
};
