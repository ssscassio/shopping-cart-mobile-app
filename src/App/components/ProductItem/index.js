/**
 * @format
 * @flow
 */
import React from 'react';
import { View, Text, Image, TouchableOpacity } from 'react-native';

import formatCurrency from '../../util';
import styles from './styles';

type Props = {
  navigate: (item: {
    id: string,
    description: string,
    value: number,
    qtd: number,
    picture: string,
    title: string,
  }) => void,
  item: {
    id: string,
    description: string,
    value: number,
    qtd: number,
    picture: string,
    title: string,
  },
};

const ProductItem = (props: Props) => {
  const { navigate } = props;
  const {
    item: { picture, value, title },
  } = props;
  return (
    <TouchableOpacity onPress={() => navigate(props.item)} style={styles.container}>
      <Image style={styles.picture} source={{ uri: picture }} />
      <View style={styles.rightContainer}>
        <Text style={styles.productTitle}>{title}</Text>
        <Text style={styles.productPrice}>{formatCurrency(value)}</Text>
      </View>
    </TouchableOpacity>
  );
};

export default ProductItem;
