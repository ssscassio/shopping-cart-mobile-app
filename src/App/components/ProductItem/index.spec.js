/** @format */

import React from 'react';
import { shallow } from 'enzyme';

import ProductItem from '.';

describe('<ProductItem />', () => {
  const wine = {
    id: 'a',
    description:
      'Análise Sensorial: \n Análise Visual: cor rosa com reflexos alaranjados. Análise Olfativa: Frutas vermelhas (morango) Análise Gustativa: seco, corpo médio, excelente acidez, taninos de boa qualidade, com retro-olfato de frutas.',
    value: 200,
    qtd: 20,
    picture: 'https://cdn.awsli.com.br/600x700/365/365010/produto/13341816/fe9a59131b.jpg',
    title: 'Vinho Nederburg Rosé',
  };

  test('should render properly', () => {
    const wrapper = shallow(<ProductItem item={wine} />);
    expect(wrapper).toMatchSnapshot();
  });
});
