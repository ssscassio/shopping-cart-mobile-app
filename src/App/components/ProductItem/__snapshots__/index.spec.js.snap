// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductItem /> should render properly 1`] = `
<TouchableOpacity
  activeOpacity={0.2}
  onPress={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "rgba(245, 245, 246, 1.00)",
      "elevation": 5,
      "flex": 1,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "marginHorizontal": 20,
      "marginVertical": 10,
      "padding": 10,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.35,
      "shadowRadius": 3,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "https://cdn.awsli.com.br/600x700/365/365010/produto/13341816/fe9a59131b.jpg",
      }
    }
    style={
      Object {
        "backgroundColor": "rgba(225, 226, 225, 1.00)",
        "borderColor": "rgba(249, 200, 34, 1.00)",
        "borderRadius": 5,
        "borderWidth": 2,
        "height": 100,
        "width": 100,
      }
    }
  />
  <View
    style={
      Object {
        "flexDirection": "column",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#333",
          "fontSize": 15,
          "fontWeight": "bold",
        }
      }
    >
      Vinho Nederburg Ros√©
    </Text>
    <Text
      style={
        Object {
          "color": "rgba(114, 171, 65, 1.00)",
        }
      }
    >
      R$ 200,00
    </Text>
  </View>
</TouchableOpacity>
`;
